File: generate-icon.js
const fs = require('fs');

// Simple SVG to PNG conversion using canvas API
function createSVG(size) {
  const scale = size / 128;
  
  return `<svg xmlns="http://www.w3.org/2000/svg" width="${size}" height="${size}" viewBox="0 0 128 128">
    <!-- Background -->
    <rect width="128" height="128" rx="20" fill="#3498db"/>
    
    <!-- Checkmark -->
    <path d="M 28 64 L 52 88 L 100 40" 
          stroke="white" 
          stroke-width="12" 
          fill="none" 
          stroke-linecap="round" 
          stroke-linejoin="round"/>
    
    <!-- Task lines -->
    <line x1="32" y1="32" x2="96" y2="32" 
          stroke="white" 
          stroke-width="8" 
          stroke-linecap="round" 
          opacity="0.6"/>
    
    <line x1="32" y1="48" x2="80" y2="48" 
          stroke="white" 
          stroke-width="8" 
          stroke-linecap="round" 
          opacity="0.6"/>
  </svg>`;
}

// Save SVG files
const sizes = [16, 32, 48, 128];

sizes.forEach(size => {
  const svg = createSVG(size);
  const filename = size === 128 ? 'icon.svg' : `icon${size}.svg`;
  fs.writeFileSync(filename, svg);
  console.log(`Generated ${filename}`);
});

console.log('\nSVG icons generated successfully!');
console.log('Note: To convert to PNG, you can use:');
console.log('1. Open generate-icon.html in a browser');
console.log('2. Or use an online SVG to PNG converter');
console.log('3. Or use Chrome extension to load the SVG directly');

File: generate_icon.py
from PIL import Image, ImageDraw
import math

def create_icon(size):
    """Create a SimpleTask icon with the given size"""
    img = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    draw = ImageDraw.Draw(img)
    
    scale = size / 128.0
    
    # Background - rounded rectangle
    bg_color = (52, 152, 219, 255)  # #3498db
    corner_radius = int(20 * scale)
    
    # Draw rounded rectangle background
    draw.rounded_rectangle(
        [0, 0, size-1, size-1],
        radius=corner_radius,
        fill=bg_color
    )
    
    # Checkmark
    check_color = (255, 255, 255, 255)
    line_width = max(2, int(12 * scale))
    
    # Checkmark points
    p1 = (int(28 * scale), int(64 * scale))
    p2 = (int(52 * scale), int(88 * scale))
    p3 = (int(100 * scale), int(40 * scale))
    
    # Draw checkmark with thick lines
    draw.line([p1, p2], fill=check_color, width=line_width)
    draw.line([p2, p3], fill=check_color, width=line_width)
    
    # Task lines (semi-transparent)
    line_color = (255, 255, 255, 153)  # 60% opacity
    line_width_small = max(1, int(8 * scale))
    
    # Line 1
    draw.line(
        [(int(32 * scale), int(32 * scale)), (int(96 * scale), int(32 * scale))],
        fill=line_color,
        width=line_width_small
    )
    
    # Line 2
    draw.line(
        [(int(32 * scale), int(48 * scale)), (int(80 * scale), int(48 * scale))],
        fill=line_color,
        width=line_width_small
    )
    
    return img

def main():
    # Generate icons in different sizes
    sizes = [16, 32, 48, 128]
    
    for size in sizes:
        icon = create_icon(size)
        if size == 128:
            # Main icon
            icon.save('icon.png', 'PNG')
            print(f"Generated icon.png ({size}x{size})")
        else:
            icon.save(f'icon{size}.png', 'PNG')
            print(f"Generated icon{size}.png ({size}x{size})")
    
    print("\nAll icons generated successfully!")

if __name__ == '__main__':
    main()
